src/components/query-viewer/query-viewer.tsx

```diff
@@ -624,15 +627,47 @@ export class QueryViewer {
 
   /** User Events for Table*/
 
-  // ToDo: implement this
-  // @Listen("TableOnItemClickEvent")
-  // handleTableOnItemClickEvent(event: CustomEvent<TableItemClick>) {}
-  // // ToDo: implement this
-  // @Listen("TableOnPageChangeEvent")
-  // handleTableOnDragundDropEvent(event: CustomEvent<TablePageChange>) {}
-  // // ToDo: implement this
-  // @Listen("TableOnFilterChangedEvent")
-  // handleTableOnFilterChangedEvent(event: CustomEvent<TableFilterChanged>) {}
+  @Listen("TableOnItemClickEvent")
+  handleTableOnItemClickEvent(event: CustomEvent<TableItemClick>) {
+    const eventData = itemClickDataForPivotTable(
+      (event as any).parameter.QueryviewerId,
+      (event as any).parameter.Data
+    );
+    this.itemClick.emit(eventData);
+  }
+
+  @Listen("TableOnPageChangeEvent")
+  handleTableOnDragundDropEvent(event: CustomEvent<TablePageChange>) {
+    let eventData;
+
+    switch ((event as any).parameter.Navigation) {
+      case PivotTableNavigation.OnFirstPage:
+        eventData = this.pivotRenderRef.firstPage();
+        this.firstPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnLastPage:
+        eventData = this.pivotRenderRef.lastPage();
+        this.lastPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnNextPage:
+        eventData = this.pivotRenderRef.nextPage();
+        this.nextPage.emit(eventData);
+        break;
+      case PivotTableNavigation.OnPreviousPage:
+        eventData = this.pivotRenderRef.previousPage();
```

Comment:
StencilJS' methods are async, so we must `await` until the method resolves.

```suggestion
  async handleTableOnDragundDropEvent(event: CustomEvent<TablePageChange>) {
    let eventData;

    switch ((event as any).parameter.Navigation) {
      case PivotTableNavigation.OnFirstPage:
        eventData = await this.pivotRenderRef.firstPage();
        this.firstPage.emit(eventData);
        break;
      case PivotTableNavigation.OnLastPage:
        eventData = await this.pivotRenderRef.lastPage();
        this.lastPage.emit(eventData);
        break;
      case PivotTableNavigation.OnNextPage:
        eventData = await this.pivotRenderRef.nextPage();
        this.nextPage.emit(eventData);
        break;
      case PivotTableNavigation.OnPreviousPage:
        eventData = await this.pivotRenderRef.previousPage();
```